### DISCUSSION TITLE
Handle path-based assetPrefix correctly in 'gatsby serve' command

### DISCUSSION THREAD

**Discussion by szimek** [gatsbyjs/gatsby#34184] [Ideas / Feature Requests]
When `assetPrefix` option is set to a path rather than a domain (e.g. `/assets/my-app`), when Gatsby builds the app it does not generate asset files in `public/assets/my-app/`, but still in `public/` directory. This causes a problem when using `gatsby serve` command, as asset files have `/assets/my-app/` prefix, and the express server can't find them.

Would it make sense to handle this case correctly by default (vs having some generic "rewrite" functionality) , i.e. add something like this to the express server used by `serve` command:
```js
// https://github.com/gatsbyjs/gatsby/blob/master/packages/gatsby/src/commands/serve.ts
const { 
  pathPrefix: configPathPrefix,
  assetPrefix: configAssetPrefix
} = config || {}
const pathPrefix = prefixPaths && configPathPrefix ? configPathPrefix : `/`
const assetPrefix = prefixPaths && configAssetPrefix && configAssetPrefix.startsWith(`/`) && !configAssetPrefix.startsWith(`//`) ? configAssetPrefix : `/`

...

router.use(assetPrefix, express.static(`public`, { dotfiles: `allow` }))
```
? The check if `assetPrefix` is a path or domain could also be done differently, e.g. with `is-url-superb` or `URL` class.

In production I can probably solve it with nginx configuration, e.g. (not tested)
```nginx
location /assets/my-app/ {
    alias /public/;
}
```
In Next.js I can solve it with `rewrites` option, e.g.:
```js
module.exports = {
  assetPrefix: '/assets/my-app',
  async rewrites() {
    return [
      { 
        source: '/assets/my-app/_next/:path*', 
        destination: '/_next/:path*' 
      }
    ]
  },
};
```

**Comment by violy** [comment]
my hacky solution : make a relative **symlink** targeting `../` in `public` directory `onPostBuild`, in `gatsby-node.js` — don't know if it works on all OS !

**Comment by szimek** [comment]
@violy Thank you! I actually even have a command that moves all static files into this new folder, but using a link is so much simpler 🤦‍♂️ I was looking for an alternative, because moving files breaks a lot of stuff (e.g. incremental builds).

Still, it would be great if it was supported out of the box.
